xpk.py

```diff
@@ -2673,6 +2698,222 @@ def workload_list(args) -> None:
+  # 1. Print out the resource flavor.
+  # 2. List what is next in Queue.
+  # 3. Split inspector into different subcommands to parse info easier.
+
+  final_return_code = 0
+  xpk_print(args)
+
+  add_zone_and_project(args)
+  set_cluster_command_code = set_cluster_command(args)
+  if set_cluster_command_code != 0:
+    xpk_exit(set_cluster_command_code)
+
+  inspector_file = write_temporary_file("==================\nXPK inspector OUTPUT:\n==================\n")
+  command_and_descriptions = [
+    ('gcloud version', 'Local Setup: gcloud version'),
+    ('gcloud config get project; gcloud config get compute/zone; gcloud config get compute/region',
+     'Local Setup: Project / Zone / Region'),
+    (f'gcloud beta container clusters list --project {args.project} --region {zone_to_region(args.zone)}'
+     f' | grep -e NAME -e {args.cluster}','GKE: Cluster Details'),
+    (f'gcloud beta container node-pools list --cluster {args.cluster}  --project={args.project} '
+     f'--region={zone_to_region(args.zone)}', 'GKE: Node pool Details'),
+    ('kubectl get node -o custom-columns=\'NODE_NAME:metadata.name,'
+     ' READY_STATUS:.status.conditions[?(@.type=="Ready")].status,'
+     ' NODEPOOL:metadata.labels.cloud\\.google\\.com/gke-nodepool\'', 'Kubectl: All Nodes'),
+    ('kubectl get node -o custom-columns=\':metadata.labels.cloud\\.google\\.com/gke-nodepool\''
+     ' | sort | uniq -c', 'Kubectl: Number of Nodes per Node Pool'),
+    ('kubectl get node -o custom-columns=\'NODE_NAME:metadata.name,'
+     ' READY_STATUS:.status.conditions[?(@.type=="Ready")].status,'
+     ' NODEPOOL:metadata.labels.cloud\\.google\\.com/gke-nodepool\' | grep -w True | awk {\'print $3\'} | sort | uniq -c',
+     'Kubectl: Healthy Node Count Per Node Pool'),
+    (f'kubectl describe ClusterQueue {_CLUSTER_QUEUE_NAME}', 'Kueue: ClusterQueue Details'),
+    (f'kubectl describe LocalQueue {_LOCAL_QUEUE_NAME}', 'Kueue: LocalQueue Details'),
+    ('kubectl describe Deployment kueue-controller-manager -n kueue-system', 'Kueue: Kueue Deployment Details'),
+    ('kubectl describe Deployment jobset-controller-manager -n jobset-system', 'Jobset: Deployment Details'),
+    ('kubectl logs deployment/kueue-controller-manager -n kueue-system --tail=100 --prefix=True', 'Kueue Manager Logs'),
+    ('kubectl logs deployment/jobset-controller-manager -n jobset-system --tail=100 --prefix=True', 'Jobset Manager Logs'),
+  ]
+
+  for command, description in command_and_descriptions:
+    return_code = inspector_run_command_helper(args, command, description, inspector_file)
+    if return_code != 0:
+      final_return_code = return_code
+      xpk_print(f'inspector failed in command: {command} description: {description} return code: {return_code}')
+
+  # Workload list views:
+  args.filter_by_job=None
+  args.filter_by_status='EVERYTHING'
+  command_description = f'xpk workload list --filter-by-status={args.filter_by_status}, --filter-by-job={args.filter_by_job}'
```

Comment:
Can we move Lines 2832-2836 to a function and re-use for `QUEUED` and `RUNNING` status?

Reply:
Created a for loop to reduce repetitiveness! Thanks Surbhi!